_resources/port1.0/group/gpg_verify-1.0.tcl

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+#
+# This portgroup is for ports that verify package-provided gpg signatures
+#
+# A single signature verification by the Portfile author is sufficient,
+# along with the standard checksum phase. The default Portfile behavior
+# is not to check gpg signatures, and assume that the Portfile author
+# has already done this. The flag `gpg_verify.use_gpg_verification`
+# is used within conditionals to run the procedure `gpg_verify.verify_gpg_signature`.
+# 
+# Usage:
+# PortGroup       gpg_verify 1.0
+# set gpg_verify.use_gpg_verification   true
+# gpg_verify.verify_gpg_signature       keyid_file signature_file test_file
+#
+# Note that many PGP key servers are intermittently accessible; therefore, it is
+# good practice to include the keyid file in ${filespath}. Also, the checksum
+# phase requires at least one hash check of downloaded files, including PGP
+# signatures for which hash checks are unnecessary. It is therefore recommended
+# to use `size` for signature file checksums, which is often constant for the same
+# keyid.
+# Example (from the julia Portfile):
+# if {${use_gpg_verification} == "true"} {
+#     distfiles-append \
+#                     ${name}-${version}-full${extract.suffix}.asc
+#     checksums-append \
+#                     ${name}-${version}-full${extract.suffix}.asc \
+#                     size    866
+# }
+
+set gpg_verify.use_gpg_verification     false
+
+set gpg_verify.gpg_homedir              ${workpath}/.gnupg
```

Comment:
Should these instead use `options`/`default`, as done for other portgroups?

Reply:
Sure, maybe? I honestly don't know how to write a portgroup and based this on a simple example without that stuff. Would you please suggest alternate code?

Reply:
Unfortunately I am not familiar with writing portgroups either. I'm not sure if I will be able to provide an alternative implementation in the near future.

I think the bigger idea is that a different structure for your proposed portgroup can make it more closely resemble other portgroups as well as be easier for ports to use. One example of this would be to make the portgroup responsible for adding a `pre-extract` phase rather than the individual ports. Maybe something resembling the `github.setup` command: `gpg_verify.setup pubkey_file signature_file test_file`.

Reply:
I also wonder which makes more sense: `pre-extract`, or `post-checksum` (or something else)?

Reply:
Thanks. I’m going to try my hand copying what @neverpanic just did with `haskell_stack` in PR https://github.com/macports/macports-ports/pull/5167 and see how it goes.

Reply:
Major mod just pushed that incorporates these comments.